cmd/kubelet/app/server.go

```diff
@@ -525,6 +526,21 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 				return err
 			}
 		}
+		// It is safe to invoke `MachineInfo` on cAdvisor before logically initializing cAdvisor here because
+		// machine info is computed and cached once as part of cAdvisor object creation.
+		info, err := kubeDeps.CAdvisorInterface.MachineInfo()
+		if err != nil {
+			return err
+		}
+		capacity := cadvisor.CapacityFromMachineInfo(info)
+		if s.KubeletConfiguration.PodsPerCore > 0 {
+			capacity[v1.ResourcePods] = *resource.NewQuantity(
+				int64(math.Min(float64(info.NumCores*int(s.KubeletConfiguration.PodsPerCore)), float64(s.KubeletConfiguration.MaxPods))), resource.DecimalSI)
+		} else {
+			capacity[v1.ResourcePods] = *resource.NewQuantity(int64(s.KubeletConfiguration.MaxPods), resource.DecimalSI)
+		}
+		capacity[v1.ResourceNvidiaGPU] = *resource.NewQuantity(int64(s.KubeletConfiguration.NvidiaGPUs), resource.DecimalSI)
```

Comment:
Resources can get added by other components on a cluster too. So from a Node Allocatable standpoint, we only want to reserve some resource off of capacity.
I feel a better approach might be that of treating the list of resources that`GetNodeAllocatable()` returns as a reservation and have kubelet_node_status.go subtract that reservation from Capacity. WDYT?
The `getNodeAllocatableInternal` is still needed for enforcement purposes.